src/components/ReportWelcomeText.tsx

```diff
@@ -1,92 +1,65 @@
-import lodashGet from 'lodash/get';
-import PropTypes from 'prop-types';
+import _ from 'lodash';
 import React from 'react';
 import {View} from 'react-native';
-import {withOnyx} from 'react-native-onyx';
-import _ from 'underscore';
+import {OnyxEntry, withOnyx} from 'react-native-onyx';
+import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
-import compose from '@libs/compose';
 import Navigation from '@libs/Navigation/Navigation';
 import * as OptionsListUtils from '@libs/OptionsListUtils';
 import * as PolicyUtils from '@libs/PolicyUtils';
 import * as ReportUtils from '@libs/ReportUtils';
-import reportPropTypes from '@pages/reportPropTypes';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type {PersonalDetailsList, Policy, Report} from '@src/types/onyx';
 import Text from './Text';
 import UserDetailsTooltip from './UserDetailsTooltip';
-import withLocalize, {withLocalizePropTypes} from './withLocalize';
-
-const personalDetailsPropTypes = PropTypes.shape({
-    /** The login of the person (either email or phone number) */
-    login: PropTypes.string,
-
-    /** The URL of the person's avatar (there should already be a default avatar if
-  the person doesn't have their own avatar uploaded yet, except for anon users) */
-    avatar: PropTypes.string,
-
-    /** This is either the user's full name, or their login if full name is an empty string */
-    displayName: PropTypes.string,
-});
-
-const propTypes = {
-    /** The report currently being looked at */
-    report: reportPropTypes,
-
-    /** The policy object for the current route */
-    policy: PropTypes.shape({
-        /** The name of the policy */
-        name: PropTypes.string,
-
-        /** The URL for the policy avatar */
-        avatar: PropTypes.string,
-    }),
-
-    /* Onyx Props */
 
+type ReportWelcomeTextOnyxProps = {
     /** All of the personal details for everyone */
-    personalDetails: PropTypes.objectOf(personalDetailsPropTypes),
-
-    ...withLocalizePropTypes,
+    personalDetails: OnyxEntry<PersonalDetailsList>;
 };
 
-const defaultProps = {
-    report: {},
-    policy: {},
-    personalDetails: {},
+type ReportWelcomeTextProps = ReportWelcomeTextOnyxProps & {
+    /** The report currently being looked at */
+    report: Report;
+
+    /** The policy for the current route */
+    policy: Policy;
 };
 
-function ReportWelcomeText(props) {
+function ReportWelcomeText({report = {} as Report, policy = {} as Policy, personalDetails = {}}: ReportWelcomeTextProps) {
```

Comment:
```suggestion
function ReportWelcomeText({report, policy, personalDetails = {}}: ReportWelcomeTextProps) {
```

Let's avoid assigning those variables to empty objects `{}`. `report` and `policy` is being used in `ReportActionItemCreated` and these props comes from Onyx, so the value can be `null` and you can treat the nullability in the code.

Reply:
Thanks, addressed by [3fad7ce](https://github.com/Expensify/App/pull/33251/commits/3fad7ceb9c9cf49242e2e42556ca5f36c296d56b)